src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -170,5 +177,95 @@ public RestClient FindRestClient(OperationGroup operationGroup)
             EnsureHeaderModels().TryGetValue(operation, out var model);
             return model;
         }
+
+        private void DecorateOperationGroup()
+        {
+            foreach (var operations in _codeModel.OperationGroups)
+            {
+                operations.ProviderName = _context.Configuration.OperationGroupMapping.ContainsKey(operations.Key) ? _context.Configuration.OperationGroupMapping[operations.Key] : ConstructOperationProviderName(operations);
+            }
+        }
+
+        private string ConstructOperationProviderName(OperationGroup operations)
+        {
+
+            string? providerName = "";
+            var request = GetBestMethod(operations);
+            bool adding = false;
+            if (request != null)
+            {
+                foreach (var segment in GetPathSegments(request.Path))
+                {
+                    var asSplit = segment.Split('/', StringSplitOptions.RemoveEmptyEntries);
+                    if (asSplit?.Length > 1 && asSplit.First().Equals("providers"))
```

Comment:
gonna guess that asSplit[0] is more efficent than asSplit.First() since we know we got back an array this is equally as safe